ape_etherscan/verify.py

```diff
@@ -300,6 +303,13 @@ def attempt_verification(self):
 
             else:
                 raise  # this error
+        except ContractVerificationError as err:
```

Comment:
Instead of excepting this and checking the message, we can move the part:

```python
if code_format == "solidity-single-file" and via_ir:
            raise IncompatibleCompilerSettingsError("via_ir", via_ir)
```

to this method instead, above this part, and fail before calling anything in `self.contract_client`.